### ISSUE TITLE
Video input device which was active is no longer available, stopping video

### ISSUE THREAD

**Issue by mnosov622** [aws/amazon-chime-sdk-js#2999]
### What happened and what did you expect to happen?

Users on iPad (10th Gen) have Front Ultra Wide Camera as selected camera, then user is then attempting to change to the standard Front Camera. Upon selecting the Front Camera, the camera will work for a moment before cutting out with their video/camera feed then going black.

Here are the errors we get in the logs:

Video input device which was active is no longer available, stopping video.

cannot bind video preview since video input device has not been chosen

could not acquire video input from current device

Issue seems be specifically to Ipads, is this known issue ? Is there a way to overcome that ?

Thank you for response.

### Have you reviewed our existing documentation?

- [X] GitHub [issues](https://github.com/aws/amazon-chime-sdk-js/issues)
- [X] FAQs [page](https://aws.github.io/amazon-chime-sdk-js/modules/faqs.html)
- [X] README [page](https://github.com/aws/amazon-chime-sdk-js#amazon-chime-sdk-for-javascript)
- [X] Feature-specific [guides](https://github.com/aws/amazon-chime-sdk-js#guides)

### Reproduction steps

N/A

### Amazon Chime SDK for JavaScript version

3.23.0

### What browsers are you seeing the problem on?

Safari (Ipad 10th)

### Browser version

safari 17 (17.7.0)

### Browser console logs

1) Video input device which was active is no longer available, stopping video.

2) cannot bind video preview since video input device has not been chosen

3) could not acquire video input from current device

4) Issue seems be specifically to Ipads, is this known issue ? Is there a way to overcome that ?

**Comment by hensmi-amazon** [comment]
This was an iOS issue that was fixed in iOS 18. For this reason we do not plan on attempting to mitigate this for iOS 17. Please reopen if it is still an issue on iOS 18.

**Comment by mnosov622** [comment]
This is still an issue on IOS 18.

**Comment by hensmi-amazon** [comment]
Got it. Must be a slightly different issue. Can you provide full logs, including WebRTC verbose logs?
![Screenshot 2024-08-26 at 2 19 19 PM](https://github.com/user-attachments/assets/4e59a54c-9c42-4c6b-b11a-320ad8e47742)

**Comment by mnosov622** [comment]
I already sent the logs to the AWS Support, let me know if you got it. If not, let me know where else I can send it privately, so I don't send it here.

**Comment by hensmi-amazon** [comment]
Got it, I should have access to those eventually. They were iOS 18 logs correct? With the verbose browser logs? It should have a lot more then logs from just the SDK.

**Comment by mnosov622** [comment]
They were IOS 17 logs from the logrocket, I don't know if you heard of this app. Will it contain verbose browser logs?

If not, could you please send instructions on how to get them ?

**Comment by hensmi-amazon** [comment]
My bad, I thought you were viewing the logs in Safari desktop app already. You can connect your phone to a MacOS device and get logs by following the steps in this blog: https://www.lifewire.com/activate-the-debug-console-in-safari-445798 . I usually have luck if all devices involved are updated to latest version.

After you have those logs working, navigate to the safari console tab in the image i sent and enable verbose logging for those other 3 settings.

**Comment by mnosov622** [comment]
Thank you, but that seems like a lot to ask from a customer to be honest, would the logs that I sent to AWS support be enough? Let me know once you get them please

**Comment by hensmi-amazon** [comment]
If you have a MacOS device it takes around 2 minutes to set up. The browser logs will not likely have enough information, since I will likely need to cut a ticket to apple. I can see if our QA team can reproduce but it may delay mitigation.

**Comment by mnosov622** [comment]
I will check if we can get that information, but hope that those logs might be useful!

**Comment by mnosov622** [comment]
Hi!

Do you have any update on this issue ?

**Comment by mnosov622** [comment]
Hi!

@hensmi-amazon I'm a bit confused with instructions. It says how to connect IOS device to Mac, and your picture shows how to enable verbose logs, but where to actually export them ?

**Comment by hensmi-amazon** [comment]
You can hold off on that for now, I think with the Javascript console logs we have a hypothesis on a possible race condition. Note that if this change went it it would be released mid December.

Regarding the instructions: Safari will add the browser internal logs to the console logs. There are instructions on export here: https://stackoverflow.com/questions/15507059/save-safari-javascript-console-to-file

**Comment by mnosov622** [comment]
Got it, thank you! So you don't need other logs for now, correct ?

**Comment by hensmi-amazon** [comment]
No logs for now, thank you!

**Comment by mnosov622** [comment]
Got it! Is this the new feature of Ipads that have Ultra Wide camera and Front Facing camera causing the main issue ?

**Comment by hensmi-amazon** [comment]
Not sure yet

**Comment by hensmi-amazon** [comment]
Maybe the internal ticket got lost. But the last thing the other engineer requested was to try to enable the verbose logs as indicated above. They also asked:

> Does the customer only see the log: "Video input device which was active is no longer available, stopping video" when the workflow fails? Or is this log message seen every time regardless of failure/success?

**Comment by mnosov622** [comment]
@hensmi-amazon  I thought you said no need for logs some time ago, so I was waiting. Yes, it happens when they start the video.

**Comment by hensmi-amazon** [comment]
The other engineer had a theory on the issue, but it turned out they still needed further logs, unfortunately it seems like the support engineer did not forward the request :/.

**Comment by mnosov622** [comment]
Is there any way to get those logs without user ? We have logs in logrocket, is that not enough ?

**Comment by mnosov622** [comment]
That error happens during initialization of the video

**Comment by mnosov622** [comment]
@hensmi-amazon we got the logs, where I can send them, don't think it's a good idea to send them here. Also, one person checked it just now, and it worked fine, was there something released ?

**Comment by michhyun1** [comment]
There wasn't anything released that addressed this - if you can reproduce it, please create a support ticket with AWS Support and paste the logs in the ticket.

**Comment by mnosov622** [comment]
I passed the logs with webRTC enabled to AWS support team as requested by @hensmi-amazon , hopefully you receive them soon! Please update me on anything that needs as well as any news regarding this issue.

**Comment by hensmi-amazon** [comment]
Hello again, just want to say sorry for the delay. Holidays and internal/external tracking were a tough combination. Thank you for the logs.

First, can you provide a code snippet for how you are calling `startVideoInput` (are you `await`ing this?) and `startVideoPreviewForVideoInput`? It seems like the latter is called before the former (which has async calls to the browser) completes.

**Comment by mnosov622** [comment]
Hi ! 

No problem, understand. Again, this works for all browsers and devices, except the iPad one, we are checking if issues is resolved, since I found the following on github:

https://github.com/aws/amazon-chime-sdk-js/issues/1059 and it says:

_In iPad devices running iOS 17, the local camera may go blank. This is fixed in iOS 18._ Am I understanding this correctly ?

We use it :

`await meetingSession.audioVideo.startVideoInput(selectedCameraId);`

**Comment by hensmi-amazon** [comment]
That was what i was initially thinking, but you mentioned that this still reproduced in iOS 18 so it must be a different issue. Can you provide the snippet that includes the call to `startVideoPreviewForVideoInput`? Is that being done right after?

**Comment by mnosov622** [comment]
I will double check for IOS 18, but did you receive the logs, were they helpful?

` meetingSession.audioVideo.startVideoPreviewForVideoInput(
            videoRef.current,
          );`

yes after

**Comment by hensmi-amazon** [comment]
Do you mean that the code is 

```
await meetingSession.audioVideo.startVideoInput(selectedCameraId);
meetingSession.audioVideo.startVideoPreviewForVideoInput( videoRef.current );
```

?

I'm asking because i see the following in the logs

```
**  - requesting new video device with constraint \{"video":\{"deviceId":\{"exact":"9FF432D16E39613A8D1AA0B8DE3C5CDCA2D57FC5"\},"width":\{"ideal":960\},"height":\{"ideal":540\},"frameRate":\{"ideal":15\}\}\} (39788c8c7bfb53cb083d.js, line 2)\**
[Log] Console-Error: wanr:  \'96 "cannot bind video preview since video input device has not been chosen" (39788c8c7bfb53cb083d.js, line 2)\
[Warning] 2024-12-18T20:22:01.931Z [WARN] OWlVideo - cannot bind video preview since video input device has not been chosen (39788c8c7bfb53cb083d.js, line 2)\
[Info] 2024-12-18T20:22:01.931Z [INFO] OWlVideo - API/DefaultDeviceController/startVideoPreviewForVideoInput "" (39788c8c7bfb53cb083d.js, line 2)\
[Info] 2024-12-18T20:22:01.931Z [INFO] OWlVideo - API/DefaultAudioVideoFacade/3c736fec-734e-4f1a-9fc5-0ca70f472713/567754c6-9d68-1c54-3233-86fd634b8fbb/startVideoPreviewForVideoInput "" (39788c8c7bfb53cb083d.js, line 2)\
[Error] Not allowed to load local resource: blob://nullhttps//st-aws-video.owlpractice.ca/estimator-v1.js.map\
[Error] Not allowed to request resource\
[Error] Cannot load blob://nullhttps//st-aws-video.owlpractice.ca/estimator-v1.js.map due to access control checks.\
[Info] 2024-12-18T20:22:02.145Z [INFO] OWlVideo - Supports SIMD: true (force: false) (39788c8c7bfb53cb083d.js, line 2)\
[Info] Running 3 iterations\'85 (83ec35a1-f904-402e-a985-6e69ddd08bb2, line 4)\
** 2024-12-18T20:22:02.417Z [INFO] OWlVideo - got video device for constraints \{"video":\{"deviceId":\{"exact":"9FF432D16E39613A8D1AA0B8DE3C5CDCA2D57FC5"\},"width":\{"ideal":960\},"height":\{"ideal":540\},"frameRate":\{"ideal":15\}\}\} (39788c8c7bfb53cb083d.js, line 2)\
** 2024-12-18T20:22:02.418Z [INFO] OWlVideo - API/DefaultDeviceController/startVideoInputDevice "9FF432D16E39613A8D1AA0B8DE3C5CDCA2D57FC5" (39788c8c7bfb53cb083d.js, line 2)\
```

The logs that start with ** are from `startVideoInput`, which seems to imply that `startVideoPreviewForVideoInput` was running during the request. When you do this during the actual meeting, does the other side also see a blank tile?

**Comment by mnosov622** [comment]
Hi, now it's a bit different, we had a person with Ipad IOS 18 version.

Here is what we found:

I joined as the therapist (on computer) first and then as the client (on iPad). When I was in the preview before entering the call, all the camera options on iPad were working fine, but once the therapist admitted the client, now the camera is not working anymore.

Once the therapist left the session, the camera on ipad works again .

What could be happening ? It seems like Safari is handling things differently ?

**Comment by mnosov622** [comment]
Hi @hensmi-amazon ,

Meeting ID: 015cb794-db63-4c17-8c26-bacc31e1df4f

Logs: Couldn't upload them here, so here is a link to download file: https://drive.google.com/file/d/10VFkY3M-PVBF7epf1ciTsZ6DOu3XGf_w/view?usp=sharing .

Let me know once you download, so I can remove that.

**Comment by hensmi-amazon** [comment]
I've downloaded, thanks

**Comment by hensmi-amazon** [comment]
Are you copy pasting the logs from the safari console? They should be extremely verbose when those are enabled :).

**Comment by mnosov622** [comment]
Yes, copied logs from safari and can confirm that the video is working fine on the link you sent above. There is no background blur enabled, any ideas what can we try ?

**Comment by hensmi-amazon** [comment]
You can try to call MediaDevice APIs and `getUserMedia` directly and pass those into the SDK.

**Comment by mnosov622** [comment]
Can you provide code example please?

**Comment by xuesichao** [comment]
```js
async function listVideoDevices() {
  try {
      const stream = await navigator.mediaDevices.getUserMedia({ video: true });
      stream.getTracks().forEach(track => track.stop());
      const devices = await navigator.mediaDevices.enumerateDevices();
      const videoDevices = devices.filter(device => device.kind === 'videoinput');

      videoDevices.forEach(device => {
          console.log(`Device ID: ${device.deviceId}`);
          console.log(`Device Label: ${device.label}`);
          console.log('-------------------');
      });

  } catch (error) {
      console.error('Error accessing media devices:', error);
  }
}

// Main
await listVideoDevices();
const stream = await navigator.mediaDevices.getUserMedia({ video: {deviceId:'<videoinput-device-id>'} });
await meetingSession.audioVideo.startVideoInput(stream);
```

**Comment by mnosov622** [comment]
Hi @xuesichao, thank you for code snippet!

I wanted to clarify on how exactly is this different from what we have ?

```
const devices = await meetingSession.audioVideo.listVideoInputDevices();

return {
            media: await meetingSession.audioVideo.startVideoInput(
              devices[0].deviceId,
            ),
            deviceId: devices[0].deviceId,
            label: devices[0].label,
          };
```

Is the main difference that we are passing an ID, but you are suggesting to pass whole stream ?

**Comment by mnosov622** [comment]
And in case we are passing videoTransformDevice, how would that look with your code suggestion ?

```
 videoTransformDevice = new DefaultVideoTransformDevice(
              logger,
              videoInputDevice,
              [videoFxProcessor],
            );
          }
          if (videoTransformDevice) {
            await meetingSession.audioVideo.startVideoInput(
              videoTransformDevice,
            );
```

Thank you!

**Comment by hensmi-amazon** [comment]
Yes, the idea of passing the whole stream is that you theoretically take the Chime SDK code out of the equation. Theoretically since https://webrtc.github.io/samples/src/content/devices/input-output/  did not have issues, there must be some minute difference in how we call browser APIs that is impacting this.

And you can pass in the stream to the transform device just the same as `startVideoInput`

**Comment by mnosov622** [comment]
@hensmi-amazon sounds good, thank you! So I'm sure about transformDevice, can you please provide a code snippet for this one ?

**Comment by mnosov622** [comment]
Ah, I see what you mean, thank you ! Will try that.

**Comment by mnosov622** [comment]
Hi @hensmi-amazon !

Just wanted to update that Front Ultra Wide Camera is working fine for Ipad, not sure if solution above fixed it, but Front Camera and Black Camera on Ipad is not working. IOS version 18.3.1 . 

Is there anything else we can try ?

**Comment by mnosov622** [comment]
It's interesting that there is a grey screen instead of user camera, but not black, could you clarify what this means ?

<img width="302" alt="Image" src="https://github.com/user-attachments/assets/20dc92dc-48fe-4fee-b97e-0e8eda3bdf45" />

**Comment by hensmi-amazon** [comment]
Gray video is very unlikely, is that just the background color? Is this 100% reproducible?

**Comment by mnosov622** [comment]
We can see grey screen and no errors, could be the background color we have for no video, but something is going on there. Do you have an Ipad where you can test this ? I can send you the link to the video app ? 

Or perhaps you have some other solutions we can try ? Thank you!

**Comment by hensmi-amazon** [comment]
Is this:
* Occurring only for iPad 10?
* Only after other changes mentioned in https://github.com/aws/amazon-chime-sdk-js/issues/2999#issuecomment-2711439639 ? Did you change any logic that switches device as well?

**Comment by mnosov622** [comment]
No, I didn't change any logic, only added what you suggested.

I'm not aware about Ipad version, but customers complaint using different OS version: 14, etc.

**Comment by mnosov622** [comment]
Here is more information, that I was able to find:

On Ipad Pro, M4 on IOS 18.2.1. Person was able to see himself on camera when he logs in for a session, however once the
session is started, he displays as a grey or black screen (and his video camera no longer functions). Interestesly, he had no video issues when using an 2017 Ipad Pro.

**Comment by hensmi-amazon** [comment]
> and his video camera no longer functions

Seems to be the key point. I'm not sure why the capturer would suddenly fail without logs. Does your application subscribe to `addDeviceChangeObserver` or similar?

**Comment by mnosov622** [comment]
To detect the device change ? Yes, we subscribe with browser device change event. Do you think we can see there if device suddenly changes or something ?

**Comment by mnosov622** [comment]
@hensmi-amazon If by any chance you have an Ipad to test this, let me know please, I will share a link to meeting, so we can check together and you can gather all the logs, etc.

**Comment by hensmi-amazon** [comment]
Since it occurs when joining a room, its probably an issue with application code. Could you send logs of this reproduction?

**Comment by mnosov622** [comment]
I think I already sent them a few times via support team

**Comment by mnosov622** [comment]
Inclduing verbose logs, etc.

**Comment by mnosov622** [comment]
One thing that comes to my mind - is maybe we can add some logs once the person is admitted, to see if something suspicious is there ?

**Comment by hensmi-amazon** [comment]
I'm not sure what difference there may be. But its worth trying the latest version. In the logs I looked at, they were using iOS 17 and iOS 15 (these are reported to backend).

**Comment by mnosov622** [comment]
Doesn't look like it do anything. I tried to replicate it with BrowserStack, here are some logs I found (attached).

[logs-Ipad-pro_11-version_18.txt](https://github.com/user-attachments/files/19390031/logs-Ipad-pro_11-version_18.txt)

it's iOS, v18.0 on Safari .

Interesting is that testing on different versions of Ipad, somewhere there is issue with camera, but somewhere there is no. I noticed that Chrome doesn't give any issues with Camera, so something in connection with Safari doesn't work well..

**Comment by mnosov622** [comment]
@hensmi-amazon I can allso see this log  - "could not acquire video input from current device"  . What does it mean ?

**Comment by mnosov622** [comment]
[logs-Ipad-pro-11_v17.txt](https://github.com/user-attachments/files/19394315/logs-Ipad-pro-11_v17.txt)

Here is IOS 17. It's right after joining a session, and in a seconds the video goes off.

**Comment by mnosov622** [comment]
Hi @hensmi-amazon , 

Is there something else do you think we can try ? I feel like I tried everything, the issue is so confusing. I don't understand if it's on your side or on our, but since other cameras are working, it's hard to tell :(

**Comment by hensmi-amazon** [comment]
In the start of those logs, I see some rapid calls of `startVideoInput` and `stopVideoInput`, could you refactor those out?
```
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:34.725Z [INFO] OWlVideo - API/DefaultAudioVideoFacade/1e5345dd-99b9-4d8c-9268-1678c0962713/266ecf24-8b0f-4d1f-ce5d-cb772fb4ec2f/listVideoInputDevices false -> [{"deviceId":"1E90606B9F603B751650B40F9167719D3A848B6E","kind":"videoinput","label":"Front Ultra Wide Camera","groupId":"CD9A79552EF360CDD4588FD0B56B1810A52BA923"},{"deviceId":"EFB45F548CBEEB06E71AFA158CE530AD12C39E8F","kind":"videoinput","label":"Back Dual Wide Camera","groupId":"6A0A34BB9406C1504505A7E29737B01B2D7FECA6"},{"deviceId":"E3CE4AD7BE90D1E24812E8C7025CAF3D385FD2BB","kind":"videoinput","label":"Back Ultra Wide Camera","groupId":"12F3211118461FED7A3629219C1DD000742935EE"},{"deviceId":"44B69A46D997058924B0F1D6402FDED98078796D","kind":"videoinput","label":"Back Camera","groupId":"3726F7C8E76A47585651561CFEA9388BE37E4295"},{"deviceId":"9FEE36B359090B82FBF6B772032BF151C6ECD231","kind":"videoinput","label":"Front Camera","groupId":"6F5808EE8D7CB3F3FDB630665FE230F2D540F570"}]
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:34.726Z [INFO] OWlVideo - API/DefaultAudioVideoFacade/1e5345dd-99b9-4d8c-9268-1678c0962713/266ecf24-8b0f-4d1f-ce5d-cb772fb4ec2f/startVideoInput "9FEE36B359090B82FBF6B772032BF151C6ECD231"
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:34.726Z [INFO] OWlVideo - requesting new video device with constraint {"video":{"deviceId":{"exact":"9FEE36B359090B82FBF6B772032BF151C6ECD231"},"width":{"ideal":960},"height":{"ideal":540},"frameRate":{"ideal":15}}}
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:34.828Z [INFO] OWlVideo - got video device for constraints {"video":{"deviceId":{"exact":"9FEE36B359090B82FBF6B772032BF151C6ECD231"},"width":{"ideal":960},"height":{"ideal":540},"frameRate":{"ideal":15}}}
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:34.828Z [INFO] OWlVideo - API/DefaultDeviceController/startVideoInputDevice "9FEE36B359090B82FBF6B772032BF151C6ECD231"
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:34.828Z [INFO] OWlVideo - API/DefaultAudioVideoFacade/1e5345dd-99b9-4d8c-9268-1678c0962713/266ecf24-8b0f-4d1f-ce5d-cb772fb4ec2f/stopVideoInput
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:34.829Z [INFO] OWlVideo - API/DefaultAudioVideoFacade/1e5345dd-99b9-4d8c-9268-1678c0962713/266ecf24-8b0f-4d1f-ce5d-cb772fb4ec2f/startVideoInput "1E90606B9F603B751650B40F9167719D3A848B6E"
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:34.829Z [INFO] OWlVideo - requesting new video device with constraint {"video":{"deviceId":{"exact":"1E90606B9F603B751650B40F9167719D3A848B6E"},"width":{"ideal":960},"height":{"ideal":540},"frameRate":{"ideal":15}}}
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:35.238Z [INFO] OWlVideo - got video device for constraints {"video":{"deviceId":{"exact":"1E90606B9F603B751650B40F9167719D3A848B6E"},"width":{"ideal":960},"height":{"ideal":540},"frameRate":{"ideal":15}}}
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:35.238Z [INFO] OWlVideo - API/DefaultDeviceController/startVideoInputDevice "1E90606B9F603B751650B40F9167719D3A848B6E"
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:35.238Z [INFO] OWlVideo - API/DefaultAudioVideoFacade/1e5345dd-99b9-4d8c-9268-1678c0962713/266ecf24-8b0f-4d1f-ce5d-cb772fb4ec2f/stopVideoInput
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:35.646Z [INFO] OWlVideo - API/DefaultAudioVideoFacade/1e5345dd-99b9-4d8c-9268-1678c0962713/266ecf24-8b0f-4d1f-ce5d-cb772fb4ec2f/startVideoInput {}
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:35.646Z [INFO] OWlVideo - requesting new video device with constraint {"video":{"streamId":"327d7407-1783-4256-8792-ad6ea01d52bc"}}
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:35.646Z [INFO] OWlVideo - using media stream 327d7407-1783-4256-8792-ad6ea01d52bc for video device
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:35.646Z [INFO] OWlVideo - got video device for constraints {"video":{"streamId":"327d7407-1783-4256-8792-ad6ea01d52bc"}}
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:35.646Z [INFO] OWlVideo - API/DefaultDeviceController/startVideoInputDevice {}
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:35.853Z [INFO] OWlVideo - API/DefaultDeviceController/startVideoPreviewForVideoInput ""
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:35.854Z [INFO] OWlVideo - API/DefaultAudioVideoFacade/1e5345dd-99b9-4d8c-9268-1678c0962713/266ecf24-8b0f-4d1f-ce5d-cb772fb4ec2f/startVideoPreviewForVideoInput ""
...
{"isEnabled":false,"strength":"low"},"backgroundReplacement":{"isEnabled":false,"backgroundImageURL":"space.jpg"}}
f3b4fa9e4765395f7267.js:3 2025-03-21T13:19:36.330Z [INFO] OWlVideo - API/DefaultAudioVideoFacade/1e5345dd-99b9-4d8c-9268-1678c0962713/266ecf24-8b0f-4d1f-ce5d-cb772fb4ec2f/startVideoInput with transform device
```

Also in terms of at least reducing the impact of this issue, is it possible for your app to filter out `Front Ultra Wide`, `Back Dual Wide` and `Back Ultra Wide`? That may mitigate the issue.

**Comment by hensmi-amazon** [comment]
I would not recommend that, sounds fragile. I would just try filtering out by name.

**Comment by mnosov622** [comment]
We can filter out, but the regular front camera behaves as front ultra wide camera when doing the following:

```

 stream = await navigator.mediaDevices.getUserMedia({
          video: { facingMode: 'user' },
          audio: false,             
        });

```

Browser thinks that default front camera is ultra wide, which is not correct. Anything we can pass there to indicate we want front camera ? Passing id won't do it in our case as it's the same error.

**Comment by hensmi-amazon** [comment]
Not sure i understand. In the logs above i see all devices enumerated. Could you not have filtered out the work 'Wide' and set `9FEE36B359090B82FBF6B772032BF151C6ECD231` in that case?

**Comment by mnosov622** [comment]
We can filter out for sure, but that won't solve the issue, since it goes blank if we set front camera.

**Comment by ltrung** [comment]
Close due to inactivity. Feel free to reopen it.
